teps/0109-better-structured-provenance-retrieval-in-tekton-chains.md

```diff
@@ -0,0 +1,502 @@
+```
+Tekton Chains will follow [TEP 0075](https://github.com/tektoncd/community/blob/main/teps/0075-object-param-and-result-types.md) and [TEP 0076](https://github.com/tektoncd/community/blob/main/teps/0076-array-result-types.md) and support the sets of json schemas and described in the documents. In this proposal, the expected formats of results are also defined.
+
+As mentioned above, the signable targets can be split into inputs (sources) and outputs (artifacts). It's important that Chains are able to determine which signable targets are inputs, and which are outputs. The signable target result names are all captured in either of the results, `ARTIFACT_INPUTS` or `ARTIFACT_OUTPUTS`, which will be two arrays storing the result names that should be captured by Tekton Chains to format in-toto provenances. `ARTIFACT_INPUTS` will be captured in `subjects` and `ARTIFACT_OUTPUTS` will be captured in `materials`.
+
+It's also beneficial that the schemas defined in Chains can follow the field requirements in in-toto attestations. Currently, the type hinting is following a readable formats that users will find it more readable, such as results being populated in [upload-pypi](https://github.com/tektoncd/catalog/blob/f4708d478ee8fac6b5b68347cde087cb7c1d1b1c/task/upload-pypi/0.1/upload-pypi.yaml) and [jib-maven](https://github.com/tektoncd/catalog/blob/6a6f3543fa14d7d840fd13d19ba4452e5e319830/task/jib-maven/0.4/jib-maven.yaml) Catalog Tasks, but these gives difficulties about standarizing signable target schemas, as each signable target's metadata can be faily different from each other. For example, a Git source usually needs a git commit and revision, which are super different from what is needed for OCI images, which need image url and its digest. In-toto attestations provide a way to unify the identifiers for multiple types of signable targets, which is described in the [Use Cases](#use-cases).
+
+Generating the objects that satisfy these schemas, from human readable provenance metadata can be done by writing provenence data using Steps in Tasks into these schema structures in TaskRuns and populate them PipelineRuns' Results, and there is also a discussion around adding a new field `provenance` in PipelineRun and TaskRun, which only allows [trusted Tasks](https://github.com/tektoncd/community/blob/2413ad70a742a6e9103c531e1c5788b2b392a7eb/teps/0091-verified-remote-resources.md#requirements) and Pipelines to generate provenance info in this field. Chains should be able to accomodate what is currently capable in TaskRun and PipelineRun results, and other potential new fields specifically for provenances.
+
+### Notes and Caveats
+
+When these well defined structures grow as richer provenance are pursued, the size of the structure can eventually grow beyond the limit of the container termination message. [TEP 0086](https://github.com/tektoncd/community/pull/521) is addressing this issue.
+
+
+## Design Details
+
+### Signable Targets
```

Comment:
What's the difference between a signable target and an output artifact?

Reply:
I'm changing all wording to "signable artifacts"